src/Processors/QueryPlan/ReadFromRemote.cpp

```diff
@@ -161,9 +166,10 @@ void ReadFromRemote::addLazyPipe(Pipes & pipes, const ClusterProxy::IStreamFacto
                 max_remote_delay = std::max(try_result.staleness, max_remote_delay);
         }
 
-        if (try_results.empty() || local_delay < max_remote_delay)
+        if (!current_settings.collaborate_with_initiator && (try_results.empty() || local_delay < max_remote_delay))
```

Comment:
Because it is too complex. `createLocalPlan` method calls `InterpreterSelectQuery` and it will be too ugly to pass ParallelReplicasCoordinator or some callback there.

Reply:
ok. Add a comment)